### PR TITLE
Finalizing askYesNo and autoReplyBroadcast template fields [pr]

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?
This PR adds a `topic` property to the properties in the `templates` object of a `GET /topics/:id` response... and also adds a `topic` property to the `message` object in a `GET /broadcasts/:id`. If said `topic` object contains an `id` property, it means that when a user receives the given template/message, their conversation topic should be updated. In a future PR, [gambit-conversations](https://github.com/dosomething/gambit-conversations) will inspect this topic property and change the `conversation.topic` accordingly if a relevant broadcast/reply is received/sent.

This can be seen in two instances for our new types:

* for an `autoReplyBroadcast`, if the `message.topic` property contains an id -- the conversation topic should be updated upon receiving the broadcast. The `autoReplyBroadcast` content type contains a `topic` reference property limited to `autoReply` entries.

* for an `askYesNo` broadcast, if user responds with a yes, upon receiving a `saidYes` reply -- the conversation topic should be updated to the `saidYesTopic` reference value. The `saidYesTopic` reference field is limited to `autoReply`, `externalPostConfig`, `photoPostConfig`, and `textPostConfig` content types.

    * Note: once an `autoReply` topic is built out (with a campaign reference), it will deprecate the `externalPostConfig` type 

Also standardizes the field names to add for when a content type may be as an outbound broadcast message: it needs to contain a single-value text named `text` and an multi-value media field named `atttachments`.

#### How should this be reviewed?
Verify expected topic properties per test entries in our new types:

* `askYesNo` - e.g. `GET /broadcasts/2X4r3fZrTGA2mGemowgiEI` -- should have a topic.id set in the `templates.saidYes` property

* `autoReplyBroadcast` - e.g. `GET /broadcasts/ =6zj3wladnGyWuMWAUYyg8c` -- should have a `topic.id` set in the `message` property

#### Any background context you want to provide?
I initially thought it'd be convenient for all of our various template fields like `saidNo` on an `askYesNo` to reference a `message` entry, so we could add attachments to any given template. In practice, this started to look like a nightmare as I began to add a few different new broadcast entires (e.g. `askYesNo`, `askAutoReply` -- where for most of these, there isn't a big need to re-use any of the `saidNo` templates. This also feels consistent with our existing `externalPostConfig`, `photoPostConfig` and `textPostConfig` entries -- as most of the templates we send within a topic will never contain attachments (which would potentially cost more 💰to send in the first place). I've updated the contentful factories accordingly to mirror the current field types per broadcast content type.

#### Relevant tickets
https://www.pivotaltracker.com/story/show/157369418

#### Checklist
- [x] Documentation added for new features/changed endpoints.
- [x] Tested on staging.

### CODE REVIEW

**Comment by aaronschachter** [comment]
The topic changes introduced aren't visible via Gambit Admin yet (they're visible via API response per the docs), but can verify expected templates for the test `askYesNo` entry I've created in https://gambit-admin-staging.herokuapp.com/broadcasts/2X4r3fZrTGA2mGemowgiEI as well as the test `autoReply` topic 

Can also verify legacy broadcasts (which seems a little pre-emptive to call legacy since none of the new broadcast content types work are supported yet, but hey) still work as expected by testing:

* `broadcast 4rBt6NOjxueEug0Iciq2Ss` - a `survey_response` broadcast (should set topic to `random` upon user replying to the broadcast

* `broadcast 5lnYtFAdkQoAQoWieYO6CK` - an `askSIgnup` campaign broadcast (should set topic to [The Talk](https://gambit-admin-staging.herokuapp.com/topics/39xu86nadqwGGyMK464C0e)

File: documentation/endpoints/broadcasts.md [hunk 1]
```diff
@@ -8,15 +8,21 @@ Name | Type | Description
...
 `name` | String | Internal name used to reference the broadcast
-`type` | String | The Contentful entry type, e.g. `autoReplyBroadcast`, `broadcast`
+`type` | String | The Contentful entry type, e.g. `askYesNo`, `autoReplyBroadcast`, `broadcast`
 `message` | Object | Contains the [outbound message content](https://github.com/DoSomething/gambit-conversations/blob/master/documentation/endpoints/messages.md) to send to user
 `message.text` | String |
 `message.attachments` | Array |
 `message.template` | String |
-`topic` | String | If set, updates the conversation topic to this string.*
-`campaignId` | Number | If set, updates the conversation topic to the given campaign's topic.*
+`message.topic` | Object | If an id property exists, its saved to the [conversation topic](https://github.com/DoSomething/gambit-conversations/blob/master/documentation/endpoints/topics.md) when broadcast is sent
```
Comment (rapala61) [inline documentation/endpoints/broadcasts.md]: I think this URL should be pointing to `gambit-campaigns`?

File: config/lib/helpers/contentfulEntry.js [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+ * field defined on the content type, which is used as a bot reply template in the topic.
+ * If the type contains a postType string property instead, its an older content type and its
+ * templates are configured via config/lib/helpers/topic. Ideally we should consolidate, but it'd
+ * take a bit of refactoring as the topic helper config contains default values for certain text
+ * fields to use if the field values are blank.
+ *
+ * A broadcastable content type currently requires a text field named "text" and attachments field
+ * named "attachments" to define content for the outbound broadcast.
+ *
+ */
```
Comment (rapala61) [inline config/lib/helpers/contentfulEntry.js]: Really nice comment

File: lib/helpers/contentfulEntry.js [hunk 3]
```diff
@@ -82,23 +82,31 @@ function getMessageTemplateFromContentfulEntryAndTemplateName(contentfulEntry, t
...
+      const topicRef = contentfulEntry.fields.saidYesTopic;
+      if (topicRef) {
+        topic.id = contentful.getContentfulIdFromContentfulEntry(topicRef);
+      }
     }
-    result[fieldName] = module.exports
-      .getMessageTemplateFromContentfulEntryAndTemplateName(messageEntry, fieldName);
+    result[fieldName] = {
+      text,
+      topic,
```
[the remark this answers is not in the document]
Comment (aaronschachter) [reply lib/helpers/contentfulEntry.js]: I agree that's cleaner. I forgot I was looking to include the summary fields in there, which I'm adding in a new branch:
```
{
  "data": {
    "id": "4nwTwvXmfuuYAGYgusGyyW",
    "name": "Test autoReplyBroadcast #1",
    "type": "autoReplyBroadcast",
    "createdAt": "2018-07-25T14:26:03.277Z",
    "updatedAt": "2018-08-08T18:07:53.433Z",
    "message": {
      "text": "I don't want to wait, for our lives to be over",
      "attachments": [],
      "template": "autoReplyBroadcast",
      "topic": {
        "id": "61RPZx8atiGyeoeaqsckOE",
        "name": "Default",
        "type": "autoReply",
        "createdAt": "2018-08-07T17:43:06.893Z",
        "updatedAt": "2018-08-08T19:33:08.519Z"
      }
    },
    "templates": {}
  }
}
```

**Pull request merged**